python/pyspark/ml/tests.py

```diff
@@ -1620,6 +1621,23 @@ def test_kmeans_summary(self):
         self.assertEqual(s.k, 2)
 
 
+class KMeansTests(SparkSessionTestCase):
+
+    def test_kmeans_cosine_distance(self):
+        data = [(Vectors.dense([1.0, 1.0]),), (Vectors.dense([10.0, 10.0]),),
+                (Vectors.dense([1.0, 0.5]),), (Vectors.dense([10.0, 4.4]),),
+                (Vectors.dense([-1.0, 1.0]),), (Vectors.dense([-100.0, 90.0]),)]
+        df = self.spark.createDataFrame(data, ["features"])
+        kmeans = KMeans(k=3, seed=1)
+        kmeans.setDistanceMeasure("cosine")
```

Comment:
minor, but why not just `kmeans = KMeans(k=3, distanceMeasure="cosine", seed=1)`?

Reply:
it was just to test that this method is working. Do you think it is better to switch to what you suggested?

Reply:
It probably a little more common to put the param in the constructor, which should be tested also.  you put `setDistanceMeasure` in `test_kmeans_param` which seems better, not a big deal though